cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriterFactory.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.stream.service.upload;
+
+import co.cask.cdap.data.stream.service.ConcurrentStreamWriter;
+import co.cask.cdap.data2.transaction.stream.StreamConfig;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.apache.twill.filesystem.Location;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * A {@link ContentWriterFactory} that creates {@link ContentWriter} that will
+ * write events to file directly.
+ */
+public final class FileContentWriterFactory implements ContentWriterFactory {
+
+  private final String accountId;
+  private final StreamConfig streamConfig;
+  private final ConcurrentStreamWriter streamWriter;
+  private final Map<String, String> headers;
+  private final Location uploadDirectory;
+
+  public FileContentWriterFactory(String accountId, StreamConfig streamConfig,
+                                  ConcurrentStreamWriter streamWriter, Map<String, String> headers) throws IOException {
+    this.accountId = accountId;
+    this.streamConfig = streamConfig;
+    this.streamWriter = streamWriter;
+    this.headers = ImmutableMap.copyOf(headers);
+
+    Location location = streamConfig.getLocation();
+    if (location == null) {
+      // Should never happen
+      throw new IllegalArgumentException("Unknown stream location for stream " + streamConfig.getName());
+    }
+    this.uploadDirectory = location.append("tmp")
```

Comment:
There is cleanup inside the ContentWriter. However, I found that might be accidentally remove all files that are still uploading. Updated the code. Please look at it again.